libraries/core/build.gradle.kts

```diff
@@ -38,6 +38,7 @@ dependencies {
     api(libs.compose.animation.core)
     api(libs.compose.runtime)
     api(libs.androidx.appcompat)
+    api(libs.kotlinx.collections.immutable)
```

Comment:
Last time we discussed it with @CherryPerry we decided not to use it. Latest release was around a [year ago](https://github.com/Kotlin/kotlinx.collections.immutable/tags) and there hasn't been any activity for a while

Reply:
@KovalevAndrey heh. I was going to discuss this later today. you beat me to it!
We might need to 'roll our own' immutable list data structure then?

I kept it as a draft due to being unsure about what approach to take.

Reply:
@KovalevAndrey I have removed the dependency and added our own implementation of immutable list. The PR is also now ready for review!